Assets/Scripts/StandardSamples/UI/Friends/UIFriendInteractionSource.cs

```diff
@@ -52,13 +59,29 @@ public virtual void OnFriendInteractButtonClicked(FriendData friendData)
         }
 
         //Should the friend UI update interaction state from this source?
-        public virtual bool IsDirty()
+        public virtual bool IsFriendsUIDirty()
+        {
+            return dirtyFlag;
+        }
+
+        public virtual void ResetFriendsUIDirtyFlag()
         {
-            return false;
+            dirtyFlag = false;
         }
 
-        public virtual void ResetDirtyFlag()
+        public virtual void MarkFriendsUIDirty()
         {
+            dirtyFlag = true;
+        }
+
+        /// <summary>
+        /// This function is called before refreshing the <see cref="UIFriendsMenu"/>.
+        /// This is an opportunity to refresh any caches or do processing work that should be done once,
+        /// instead of processed for each call of <see cref="GetFriendInteractionState(FriendData)"/>.
+        /// </summary>
+        public virtual void ProcessInformationBeforeFriendsRefresh()
```

Comment:
Is the intent of this virtual function being defined with an empty implementation that the _default_ implementation should be empty, and overridden by classes that derive from `UIFriendInteractionSource`?

If so, do any of the derivative classes require an empty implementation?

Reply:
Yes. This is a timing window for derivative classes to do optional processing work. There's no work that can be done in the parent class. I don't think it needs to be `abstract`, because implementors can also choose to not implement anything.

Currently, only the `UISessionsMatchmakingMenu` overrides this function, because it has work that needs to be done at this time. (before starting to process friends UI states)